IOCTL_BUSENUM_ISDEVPLUGGED: Added a new IRP Handler

shauleiz opened the pull request:
## Handler:

In file **busenum.c**
In function **Bus_IoCtl()**
Calls new function **Bus_IsDevicePluggedIn()**
## Bus_IsDevicePluggedIn()

Receives an in/out buffer. Size=ULONG
When buffer is input: Points to SerialNo
When buffer is Output: Points to 0x00000001 or 0x00000000 - signifies exists/does not exist
## Additional Info

IOCTL_BUSENUM_ISDEVPLUGGED is defined in file ScpVBus.h
## Usage

```
buffer[0] = UserIndex;      
if (DeviceIoControl(g_hBus, IOCTL_BUSENUM_ISDEVPLUGGED, buffer, _countof(buffer), output, 4, &trasfered, nullptr))
        {
            if (*output != 0)
                out = TRUE;
                        else
                out = FALSE;
        };
```

shauleiz committed:
Added Project vXboxInstall

Proof of consept - plug/unplug virtual Xbox devices

shauleiz committed:
Added Project vXboxInstall

Proof of consept - plug/unplug virtual Xbox devices

shauleiz committed:
Merge branch 'master' of https://github.com/shauleiz/ScpVBus

shauleiz committed:
Merge branch 'master' of https://github.com/nefarius/ScpVBus

shauleiz committed:
Merge branch 'master' of https://github.com/nefarius/ScpVBus

Conflicts:
	XOutput/XOutput.h

shauleiz committed:
Merge branch 'master' of https://github.com/nefarius/ScpVBus

# By Benjamin Höglinger
# Via Benjamin Höglinger
* 'master' of https://github.com/nefarius/ScpVBus:
  Update README.md
  Create README.md
  Delete ReadMe.txt
  Moved bus initialization from DllMain to call-once routine Introduced macros for repetitive tasks
  Fixed memory leak
  Fixed handling the return value of DeviceIoControl obeying MSDN

shauleiz committed:
Merge branch 'master' of https://github.com/nefarius/ScpVBus

shauleiz committed:
New Command Code: IOCTL_BUSENUM_ISDEVPLUGGED

Check that a vXbox device is plugged in

shauleiz committed:
Merge branch 'master' of https://github.com/nefarius/ScpVBus

shauleiz committed:
Merge branch 'master' of https://github.com/nefarius/ScpVBus

shauleiz committed:
Merge branch 'master' of https://github.com/shauleiz/ScpVBus

Conflicts:
	.gitignore

shauleiz committed:
Ignore List

shauleiz committed:
New Command Code: IOCTL_BUSENUM_EMPTY_SLOTS

Get the number of empty slots on the Virtual Bus

Added Comments

Reverse some unwanted changes

Removed test project

nefarius commented:
Nice! But I'd like to see the addition of the `vXboxInstall` project in a separate pull request to keep things organized. Especially since it seems to contain a lot of copy&paste which could be avoided by referencing the `XOutput` LIB.

shauleiz commented:
This is an ugly Pull Request I presented.
Project vXboxInstall was used for testing and should not have been added.
If it is not too much to ask - please merge it then remove the project.
If you don't want to, then I'll prepare a new Pull-Request

I am already working on a new IRP handler. So I can add them both simultaneously.

shauleiz committed:
XOutput: Added 2 interface functions

1. XOutputNumEmptyBusSlots() - Get the number of empty Virtual Bus slots
2. XOutputIsCtrlExist() - Checks if a device is plugged in.
